engines/sci/graphics/palette.h

```diff
@@ -111,9 +159,44 @@ class GfxPalette : public Common::Serializable {
 	bool colorIsFromMacClut(byte index);
 
 #ifdef ENABLE_SCI32
+private:
+	// SCI2 (SQ6) defines 10 cyclers
+	PalCycler *_cyclers[10];
+	/**
+	 * The cycle map is used to detect overlapping cyclers. According to SCI engine code, when two cyclers overlap,
+	 * a fatal error has occurred and the engine will display an error and then exit.
+	 */
+	bool _cycleMap[256];
+	inline void _clearCycleMap(uint16 fromColor, uint16 numColorsToClear);
+	inline void _setCycleMap(uint16 fromColor, uint16 numColorsToClear);
+	inline PalCycler *_getCycler(uint16 fromColor);
+
+	/**
+	 * The fade table records the expected intensity level of each pixel in the palette that will be displayed on
+	 * the next frame.
+	 */
+	byte _fadeTable[256];
+
+public:
+	void applyAllCycles();
+	void applyCycles();
+	void applyFade();
+
 	bool loadClut(uint16 clutId);
 	byte matchClutColor(uint16 color);
 	void unloadClut();
+
+	int16 setCycle(uint16 fromColor, uint16 toColor, int16 direction, int16 delay);
+	void doCycle(uint16 fromColor, int16 speed);
+	void cycleOn(uint16 fromColor);
+	void cyclePause(uint16 fromColor);
+	void cycleAllOn();
+	void cycleAllPause();
+	void cycleOff(uint16 fromColor);
+	void cycleAllOff();
+
+	void setFade(uint8 percent, uint16 fromColor, uint16 toColor);
+	void fadeOff();
```

Comment:
I've discussed these with m_kiewitz as well. Please, create a new GfxPalette32 class for SCI32 games, which inherits from GfxPalette and defines the extra methods needed for SCI32. This is a cleaner approach, as it separates the SCI32 code from the SCI16 code, and we've used it in other parts of the engine as well.